pyiron_base/job/worker.py

```diff
@@ -207,3 +207,36 @@ def run_static(self):
 
         # The job is finished
         self.status.finished = True
+
+    def wait_for_worker(self):
+        """
+        Wait for the workerjob to finish the execution of all jobs. If no job is in status running or submitted the 
+        workerjob shuts down automatically after 10 minutes. 
+        """
+        finished = False
+        j = 0
+        log_file = os.path.join(self.working_directory, "process.log")
+        pr = self.project_to_watch
+        master_id = self.job_id
+        while not finished:
+            df = pr.job_table()
+            df_sub = df[
+                ((df["status"] == "submitted") | (df.status == "running")) &
+                (df["masterid"] == master_id)
+            ]
+            if len(df_sub) == 0:
+                j += 1
+                if j > 10:
```

Comment:
If I read right this is a safety feature so you need consecutive blank calls before really considering yourself done. In this case 10 should be a kwarg, and it should probably be in units of seconds and related to the other new kwarg